File: Week1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>



int main(void)
{
    int height;


    do
    {
        // Asking user for the height.
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8); // Height lower than 1 and bigger than 8 is not permitted.


    int items = (height * 2) + 2;
    int gap = 2;
    int hashes;
    int spaces;


    for (int i = 0; i < height; i++)
    {
        // number of spaces per row.
        spaces = height - i - 1;

        // number of hashes per row.
        hashes = height - spaces;


        for (int j = 0; j < spaces; j++)
        {
            printf(" ");
        }


        for (int j = 0; j < hashes; j++)
        {
            printf("#");
        }


        for (int j = 0; j < gap; j++)
        {
            printf(" ");
        }


        for (int j = 0; j < hashes; j++)
        {
            printf("#");
        }


        printf("\n");
    }
}

File: Week6/sentimental-mario-more/mario.py
import cs50


height = -1

while True:
    # get input
    height = cs50.get_int("Height: ")

    # check input
    if height >= 1 and height <= 8:
        break

# declare parameters
items = (height * 2) + 2
gap = 2
spaces = 0
hashes = 0

# print pyramid
for i in range(height):
    spaces = height - i - 1
    hashes = height - spaces

    for _ in range(spaces):
        print(" ", end="")

    for _ in range(hashes):
        print("#", end="")

    for _ in range(gap):
        print(" ", end="")

    for _ in range(hashes):
        print("#", end="")

    print()

